src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+
+
+class IOFileType(enum.Enum):
+    CSV = "csv"
+    JSONL = "jsonl"
+    XLSX = "xlsx"
+
+
+def validate_input_questions(input_questions: pathlib.Path):
+    """
+    Ensure input questions file exists, is not a directory
+    and is a '.jsonl' file.
+    Args:
+        input_questions (Path): The path to the input questions .jsonl file
+    Returns:
+        None
+    """
+    if not os.path.exists(input_questions):
+        raise ValueError(
+            f"Input questions file {input_questions} does not exist",
+        )
+
+    if os.path.isdir(input_questions):
+        raise ValueError(
+            f"Input questions file {input_questions} is a directory",
+        )
+
+    # need to make sure input questions is a jsonl file
+    if input_questions.suffix.lstrip(".") != IOFileType.JSONL.value:
+        raise ValueError(
+            f"Invalid file type: {input_questions}. Expected a '.jsonl' file."
+        )
+
+
+def validate_output_file_formats(file_formats: List[str]):
+    """
+    Validates that file formats passed in for the file containing scores and
+    responses is a valid output format. Valid file formats are one of:
+    jsonl, xlsx, or csv.
+    Args:
+        file_formats (List[str]): A list of all of the file format strings
+                                  passed in by the user.
+    Returns:
+        None
+    """
+    for file_format in file_formats:
+        valid_formats = {item.value for item in IOFileType}
+        if file_format not in valid_formats:
+            raise ValueError(
+                f"File format {file_format} is not a valid output format. Format must be one of {valid_formats}"
+            )
+
+
+def create_results_file_name(
+    file_format: str, output_dir: str, timestamp: str, model_name: str
+) -> str:
+    """
+    Validates that file formats passed in for the file containing scores and
+    responses is a valid output format. Valid file formats are one of:
+    jsonl, xlsx, or csv.
+    Args:
+        file_formats (str): The file format for the output file. One of "csv",
+                            "xlsx", "jsonl".
+        output_dir (str):   Output directory for results and  scores file
+        timestamp (str):    timestamp in .iso format that is part of the
+                            name of the output file.
+        model_name (str):   Name of model to be used in the name of the output
+                            file.
+    Returns:
+        str: The file name containing the scores and model responses
+    """
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    elif IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+    elif IOFileType.JSONL.value == file_format:
+        file_type = "jsonl"
+    else:
+        raise ValueError("File format is not one of: csv, xlsx, jsonl")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.DK_BENCH_DATA_DIR
```

Comment:
What if the user's directory in the config is different from the default? Let's avoid handling this case, and instead assume that the correct value was passed down and present in `output_dir`